app/middlewares/device.ts

```diff
@@ -0,0 +1,37 @@
+import * as restify from 'restify';
+import { IRequest, IResponse } from '../interfaces/utils/IServer';
+import { IDevice } from '../interfaces/models';
+import { UnauthorizedError, InternalServerError } from 'restify-errors';
+import IMiddleware from '../interfaces/utils/IMiddleware';
+import Device from '../models/device.model';
+
+export default class AuthDeviceCheckMiddleware implements IMiddleware {
+  public async init(req: IRequest, res: IResponse, next: restify.Next) {
+    const device = <IDevice> await Device.findOne({
```

Comment:
You need to find all the devices user belongs to, A user can set more than 1 authorised devices. I think you should make it `Device.find(`